exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java

```diff
@@ -671,8 +674,9 @@ private HoldingContainer visitBooleanAnd(BooleanOperator op,
       HoldingContainer out = generator.declare(op.getMajorType());
 
       JLabel label = generator.getEvalBlockLabel("AndOP");
-      JBlock eval = generator.getEvalBlock().block();  // enter into nested block
-      generator.nestEvalBlock(eval);
+      JBlock eval = new JBlock();
```

Comment:
How about we add the following two methods in ClassGenerator.java: 

```java
  private JBlock createInnerBlock(BlockType type) {
    final JBlock currBlock = getBlock(type);
    final JBlock innerBlock = new JBlock();
    currBlock.add(innerBlock);
    return innerBlock;
  }

  protected JBlock createInnerEvalBlock() {
    return createInnerBlock(BlockType.EVAL);
  }
```

Then, replace `generator.getEvalBlock().block()` with `generator.createInnerEvalBlock()`  ?